### ISSUE TITLE
Having trouble styling Textext

### ISSUE THREAD

**Issue by eric-hu** [alexgorbatchev/jquery-textext#18]
I'm following the examples trying to call TextExt on a textarea, but I can't seem to change the styling.  I've narrowed down the issue to this CSS style on the div.text-wrap:

```
element.style {
    height: 9px;    
    width: 112px;
}
```

and also this style on the textarea inside of div.text-wrap (that is, the original textarea I defined):

```
element.style {
    width: 112px;
}
```

I can't figure out which particular line sets these attributes in the generated js.  Renaming the htmlWrap eliminated the those styles on textarea, but the tags and dropdowns disappeared as well.

```
$('#invitee_login').textext( {
    ...
    htmlWrap: '<div class="autocomplete-core"><div class="autocomplete-wrap"/></div>',
    ...
})
```

I guess I'm looking for a way to pick and choose which of the default styles are used.  Is there a way to do this through textext params?

(I apologize for using the issues tracker for a usage question--I couldn't find another resource like a mailing list for this.)

**Comment by alexgorbatchev** [comment]
@eric-hu What are you trying to accomplish? The styles you are referring to are applied to elements during run time to keep the visual dimensions in check. If you try to set those some place else, they will be overwritten.

**Comment by eric-hu** [comment]
I'm trying to build a textarea with a `width: 100%` style defined on it.  Applying TextExt to the textarea sets the fixed width style from my first post.

I haven't found a simple way for our designer to change that, and that's what my goal is.

**Comment by alexgorbatchev** [comment]
@eric-hu were you able to resolve this?

**Comment by eric-hu** [comment]
Hi Alex.  Yes, the issue was actually that I was loading TextExt into a Fancybox modal.

I was loading it before rendering the Fancybox content, and Fancybox seems to calculate some sizes only after objects are rendered into it.  I'm not 100% on the details since another dev I work with figured it out, but I'd be willing to look into it further if you're interested

**Comment by eric-hu** [comment]
I'm looking at the code for this problem and it looks like our other dev fixed the styling issue by rendering the form elements with inline styles:

```
    %textarea{name: 'invitee_logins', class: 'invitee_logins', data_display: "invitee", rows:1, style:'width:100%'}
```

That line above is written in Haml.

We have another issue with autocomplete behaving oddly and I suspect that this is all related to loading TextExt into a Fancybox 2.0 modal.  If anyone else comes up with this issue as well, I'd be happy to spend some time trying to produce a minimum demo case.

**Comment by thejae** [comment]
Hi there am having problem when I set width to 100%, the plugin defaults to 12px.

**Comment by thejae** [comment]
nvm, i replaced my textext with chosen... 2hrs implementation vs 2 days for textext...

**Comment by JoeKueser** [comment]
Anyone ever come up with a solution for this. No matter what I try, TextExt is setting the widths to 12px, which doesn't look so great ;-)

For what it's worth, my textarea is rendered into a Twitter Bootstrap style form.

**Comment by m-misseri** [comment]
I have the same problem mentioned by @JoeKueser

**Comment by funkyeah** [comment]
This is a problem with the TextExt core. It wraps the input or textarea element with divs when TextExt is activated. These divs (which have classes text-wrap and text-core) are given widths. My understanding is that the widths applied to the wrapping divs is calculated from the width of the textarea or input element TextExt is applied to. The issue is that if your input or text area element has a percentage width, this calculation does not work.

In my case, once I switched to a fixed width for the input box the issue went away. Clearly a bug IMO though.

**Comment by gjlondon** [comment]
Confirming what @funkyeah said, adding a fixed width to the textarea (e.g. in the CSS) does fix the issue for me.

**Comment by ghost** [comment]
Yes, great plugin, but please fix this!

**Comment by vsueki** [comment]
To manipulate the width of the textarea @1.3.1 i made a workaround editing the line 973 of the core.js
<code>width     = self.originalWidth + 'px',</code>

In my case i wrote
<code>width     = '100%',</code>

This is what are you trying to fix ?

**Comment by rubenve** [comment]
I know this is an old thread, but I walked into the same problem. I used vsueki's solution above, but I'm not sure what will happen in cases where you are actually using absolute widths.
